### PR TITLE
display landmark assessments hostory in frontend

### PR DESCRIPTION
**Pull request by ada-only-one-ada**
Screenshot 2024-10-03 at 10 48 06

### CODE REVIEW

File: src/components/LandmarkDrawer.tsx [hunk 1]
```diff
@@ -79,22 +92,93 @@ function Editor() {
...
+    createdAt: string;
+    score: number;
+    markdown: string | null;
+  }) => {
+    const newAssessment: LandmarkAssessment = {
+      ...assessment,
+      createdAt: new Date(assessment.createdAt) 
+    };
+    setSelectedAssessment(newAssessment);
+  };
```
Comment (weihanwang) [inline src/components/LandmarkDrawer.tsx:115]: Why do you need this logic? Why can't simply call `setSelectedAssessment(assessment)`?

File: src/components/LandmarkDrawer.tsx [hunk 2]
```diff
@@ -79,22 +92,93 @@ function Editor() {
...
+        <Tr key={index} onClick={() => handleSelectAssessment(assessment)} cursor="pointer">
+          <Td>{prettifyDate(assessment.createdAt)}</Td>
+          <Td>{assessment.score}</Td>
+          <Td>假评估人</Td>
+          <Td>
+            <Text>
+              {assessment.markdown && assessment?.markdown.length > maxChar 
+              ? `${assessment.markdown.substring(0, maxChar)}...`
+              : assessment.markdown}
+            </Text>
```
Comment (weihanwang) [inline src/components/LandmarkDrawer.tsx:140]: There is a CSS style that truncates text automatically.

File: src/components/LandmarkDrawer.tsx (hunk 1 above)
Comment (ada-only-one-ada) [reply src/components/LandmarkDrawer.tsx:115]: Type string is not assignable to type Date:
Screenshot 2024-09-30 at 17 08 04

Comment (weihanwang) [reply src/components/LandmarkDrawer.tsx:115]: It's a known problem. Use z.coerce.string().optional() as the type of zLandmarkAssessment.score and modify all impacted code. Refer to other places where coerce is used, for example, at src/shared/ChatMessage.ts:8

File: src/components/LandmarkDrawer.tsx [hunk 3]
```diff
@@ -79,22 +91,65 @@ function Editor() {
...
+    <ModalHeader>历史评估结果</ModalHeader>
+    <ModalBody>
+      <VStack gap={componentSpacing} align="left">
+        <p><b>日期：</b>{assessment.createdAt && prettifyDate(assessment.createdAt)}</p>
+        <p><b>结果：</b>{assessment.score}</p>
+        <p><b>评估人：</b>假评估人</p>
+        <p><b>详情：</b>{assessment.markdown || "无"}</p>
+      </VStack>
+    </ModalBody> 
+  </ModalContent>
```
Comment (weihanwang) [inline src/components/LandmarkDrawer.tsx:153]: Add the standard model button "Close"

Comment (ada-only-one-ada) [reply src/components/LandmarkDrawer.tsx:155]: Screenshot 2024-10-03 at 10 42 12

File: src/components/LandmarkDrawer.tsx [hunk 4]
```diff
@@ -79,22 +91,67 @@ function Editor() {
...
+function LandmarkAssessmentHistory({ landmark } : {
+  landmark: Landmark;
+}) {  
+  const [user] = useUserContext();
+  const { data: assessments } = trpcNext.map.listLandmarkAssessment.useQuery({
+    userId: user.id,
+    landmark: landmark.名称,
+  });
+
+  const [selectedAssessment, setSelectedAssessment] = useState<LandmarkAssessment>();
```
Comment (weihanwang) [inline src/components/LandmarkDrawer.tsx:103]: wrap all the lines in this file that are > 80 chars long

**Review by weihanwang — APPROVED** [review]

**Pull request merged**